src/main/java/echoserver/EchoServer.java

```diff
@@ -0,0 +1,60 @@
+package echoserver;
+
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.Scanner;
+
+public class EchoServer implements Runnable {
+    private Scanner input;
+    private PrintWriter output;
+    private Socket clientSocket;
+
+    public EchoServer(Socket clientSocket) {
+        this.clientSocket = clientSocket;
+    }
+
+    public void run() {
+        try {
+            setServerInputOutput(clientSocket);
+            output.println("Connected to server");
+            echo();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void echo() {
+        String inputString;
+        while ((inputString = SocketIO.readFromInputStream(input)) != null) {
+            if (InputValidator.isQuit(inputString)) {
+                close();
+                break;
+            }
+            SocketIO.writeToOutputStream(output, inputString);
+        }
+    }
+
+    private void setServerInputOutput(Socket clientSocket) throws IOException {
```

Comment:
The following three methods seem kinda overkill to abstract into private methods and actually makes it a little more difficult to read since there's some levels of indirection. We could just simplify this and assign these properties in the constructor like so:

```java
public EchoServer(Socker clientSocket) {
  this.clientSocket = clientSocket;
  this.input = SocketIO.createSocketReader(clientSocket);
  this.output = SocketIO.createSocketWriter(clientSocket);
}
```